doc/mgmt/Developer Guide.md

```diff
@@ -655,404 +646,1854 @@ $ tools/xfmr/annotate.sh
+    fields            []string //list of fields(container, leaf/leaflist).List is not allowed.
+    //following attributes are for transformer infra use only
+    fieldsFillAll     bool  //flag to fill all fields in container/list
+    allowFieldsXpath  map[string]bool   //proceed further only if xpath is present in this set
+    tgtFieldsXpathMap map[string][]string   //process & fill data only if xpath is present in this map.
+}
+```
+
+
+
+- Depth:
+  For processing the depth QP, the inParams.queryParams.curDepth will be referenced by the subtree.
+  The value in the curdepth indicates how much depth is yet to be traversed by the subtree.
+
+  Eg: 
+  If the curDepth is 3, the subtree needs to fill the ygot upto 3 levels of depth in the yang subtree.
+  The curDepth value is decremented by 1 for every level processed and once the curDepth value becomes 0, 
+  the subtree can stop the GET processing for subtree below it.
+
+  
+
+- Content:
+  For processing the content QP, the inParams.queryParams.content will be used.
+  The subtree shall fill the ygot based on the content field in the yang subtree
+  Refer the below table for supported content types and what needs to filled by the subtree
+
+  | Value       | Description                                                  |
+  | ----------- | ------------------------------------------------------------ |
+  | config      | Populate only configuration descendant data nodes (RW nodes) |
+  | non config  | Populate only non-configuration descendant data nodes (RO nodes) |
+  | operational | Populate data in “non-config” & not in “config”.(applicable only for OC yang) (RO nodes) |
+  | all         | Populate all descendant data nodes (RW and RO nodes)         |
+
+  
+
+- Fields:
+  For processing the fields QP, the inParams.queryParams.fields can be referenced by the subtree.
+  The fields will have the list of relative xpath of the fields that needs to be filled by the subtree. 
+  The relative xpath is in reference to the requestUri.
 
-    // requestUri holds the resource URI specified in client app.
-    // It can be pointing to parent or child node of the node at which
-    // this callback is registered. -- (RO)
-    requestUri string
 
-    // uri contains path to the node where a callback gets invoked.
-    // Functions should use this value to identify target resource. -- (RO)
-    uri string
 
-    // ygRoot is the ygot struct initialized by translib for the requestUri. -- (RW)
-    ygRoot *ygot.GoStruct
+###### 2.5.8.2 Example to support query parameters
 
-    // oper indicates the current operation requested by translib. -- (RO)
-    // 1=GET, 2=CREATE, 3=REPLACE, 4=UPDATE, 5=DELETE
-    oper int
+Example implementation of subtree callbacks using query parameters can be seen below. Note that the data trees not annotated with subtree callbacks are implicitly handled by transformer core, that it does not require codes to support query parameters.
```

Comment:
That last clause should be "and they do not require additional code to support query parameters."